website/en/docs/lang/subtypes.md

```diff
@@ -136,3 +136,37 @@ type Func2 = (number, string) => void;
 let func1: Func1 = (a: number) => {};
 let func2: Func2 = func1;
 ```
+
+#### Subtypes of unions of functions <a class="toc" id="toc-subtypes-of-unions-of-functions" href="#toc-subtypes-of-unions-of-functions"></a>
+
+A function is a subtype of another when the input is contravariant (_less specific type / supertype_) and the output is covariant (_more specific type / subtype_).
+
+```js
+// @flow
+declare var foo1: (string => ?string) => void
+declare var bar1: ?string => string
+
+foo1(bar1) // Works!
```

Comment:
This example gets straight to the point, but I think a concrete example would be better at building intuition.

A lot of javascript developers are familiar with callback functions, so how about examples using that?

```
function f(callback: (name: string) => void) {
  callback("Flow");
}

function g(name: ?string): void {
  let name = name || "World";
  console.log("Hello, " + name);
}

f(g); // `?string => void` is a subtype of `string => void`
```

```
function f(callback: () => ?string) {
  var name = callback() || "World";
  console.log("Hello, " + name);
}

function g(): string {
  return "Flow";
}

f(g); // `() => string` is a subtype of `() => ?string`
```